### ISSUE TITLE
oneflow.argsort perform differently between cpu and cuda when dim=1 

### ISSUE THREAD

**Issue by Redmept1on** [Oneflow-Inc/oneflow#10476]
## Summary

oneflow.argsort perform differently between cpu and cuda when dim=1, cuda=pytorch, cpu not

## Code to reproduce bug

```python
import oneflow as flow
import numpy as np

x1 = flow.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
x1 = x1.cuda()
y1 = flow.argsort(x1,dim=1)
print(y1)

x1 = flow.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
x1 = x1.cpu()
y2 = flow.argsort(x1,dim=1)
print(y2)

```
![image](https://github.com/Oneflow-Inc/oneflow/assets/84373897/cbb4dd8b-b643-4728-ba66-ddc5e7a4c1e6)

pytorch
```python
import torch
import numpy as np

input_tensor = torch.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
output_tensors = torch.argsort(input_tensor,dim=1)

print(output_tensors)
```
![image](https://github.com/Oneflow-Inc/oneflow/assets/84373897/452db839-0eb6-4faf-adf2-8b24e76a01e0)

## System Information

- What is your OneFlow installation :dockerhub
- OS:Linux
- OneFlow version (run python3 -m oneflow --doctor):0.9.1.dev20231024+cu117
- Python version: 3.8

**Comment by Redmept1on** [comment]
oneflow.sort has same bug

**Comment by Redmept1on** [comment]
oneflow.topk has the same bug
```python
import oneflow as flow
import numpy as np

x1 = flow.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
x1 = x1.cuda()
y1 = flow.topk(x1,k=2,dim=1)
print(y1)

x1 = flow.tensor(np.array([[float('inf'), 0, -1, float('nan'), 5]], dtype=np.float32))
x1 = x1.cpu()
y2 = flow.topk(x1,k=2,dim=1)
print(y2)
```
![image](https://github.com/Oneflow-Inc/oneflow/assets/84373897/3e3d40cd-3d17-4f91-951e-42168233a435)
